### PR TITLE
[BlobDB] Return write error on reaching blob dir size limit

### PR DESCRIPTION
**Pull request by sagar0**
I found that we continue accepting writes even when the blob db goes beyond the configured blob directory size limit. Now, we return an error for writes on reaching `blob_dir_size` limit and if `is_fifo` is set to false. (We cannot just drop any file when `is_fifo` is true.)

Deleting the oldest file when `is_fifo` is true will be handled in a later PR. 

Test Plan:
Fixed unit tests.

### CODE REVIEW

File: utilities/blob_db/blob_db_impl.cc [hunk 1]
```diff
@@ -1050,6 +1050,14 @@ uint64_t BlobDBImpl::ExtractExpiration(const Slice& key, const Slice& value,
 Status BlobDBImpl::AppendBlob(const std::shared_ptr<BlobFile>& bfile,
                               const std::string& headerbuf, const Slice& key,
                               const Slice& value, std::string* index_entry) {
+  auto size_put = BlobLogRecord::kHeaderSize + key.size() + value.size();
+  if (bdb_options_.blob_dir_size > 0 &&
+      (total_blob_space_.load() + size_put) > bdb_options_.blob_dir_size) {
+    if (!bdb_options_.is_fifo) {
+      return Status::IOError("Blob DB reached the maximum configured size.");
```
Comment (yiwu-arbug) [inline utilities/blob_db/blob_db_impl.cc]: Should return NoSpace.

**Comment by yiwu-arbug** [comment]
Why is_fifo = true is special?

**Comment by sagar0** [comment]
`is_fifo` option is not currently used anywhere in the blob_db code. And the comments above `is_fifo` and `blob_dir_size` options gave me the impression that older files could be evicted on out-of-space only if this option is:  sethttps://github.com/facebook/rocksdb/blob/386a57e6ef08fbcc2f9b92217fa4a8c9873386c9/utilities/blob_db/blob_db.h#L39-L45

File: utilities/blob_db/blob_db_impl.cc (hunk 1 above)
Comment (sagar0) [reply utilities/blob_db/blob_db_impl.cc]: good point, will fix.

File: utilities/blob_db/blob_db_test.cc [hunk 2]
```diff
@@ -664,6 +664,7 @@ TEST_F(BlobDBTest, GCOldestSimpleBlobFileWhenOutOfSpace) {
   Options options;
   options.env = mock_env_.get();
   BlobDBOptions bdb_options;
+  bdb_options.is_fifo = true;
```
Comment (sagar0) [inline utilities/blob_db/blob_db_test.cc]: This test, BlobDBTest.GCOldestSimpleBlobFileWhenOutOfSpace, already fails with the new code I added. And I think that is right ... the test shouldn't have passed in the first place as the size is going beyond the configured  `blob_dir_size` size.

**Comment by yiwu-arbug** [comment]
Ouch, I didn't even notice is_fifo is not implemented..

**Pull request closed without merging**